modules/top-level/plugins/mk-plugin-pack.nix

```diff
@@ -35,18 +35,28 @@ let
       paths = overriddenPlugins;
       inherit pathsToLink;
 
-      # Remove empty directories and activate vimGenDocHook
-      # TODO: figure out why we are running the `preFixup` hook in `postBuild`
+      # buildEnv uses runCommand under the hood. runCommand doesn't run any build phases.
+      # To run custom commands buildEnv takes postBuild argument.
+      # fixupPhase is used for propagating build inputs and to trigger vimGenDocHook
       postBuild = ''
         find $out -type d -empty -delete
-        runHook preFixup
+        fixupPhase
```

Comment:
Ah, this is the change that was alluding me. Stdenv creates the `propagated-build-inputs` file in the `fixupPhase`, so when I was collecting the `propagatedBuildInputs` if was having no effect because `fixupPhase` wasn't being run.

I'm slightly confused why my second attempt ([manually writing the `propagated-build-inputs` file](https://github.com/nix-community/nixvim/pull/3099/files#diff-69cce1aae191cd61ffbf31244f21b6fb71aa5dce7ff2fb8f9eb78500cbf54239R46-R48)) didn't work though. Maybe some of the tests needed additional work? Or maybe `_addToLuaPath` isn't as recursive as I thought?